utils/aws_api.py

```diff
@@ -0,0 +1,369 @@
+                if self.resource_has_special_tags(account, type, b, tags):
+                    continue
+                unfiltered_buckets.append(b)
+            except Exception as e:
+                if 'NoSuchTagSet' in e.message:
+                    unfiltered_buckets.append(b)
+        return unfiltered_buckets
+
+    def custom_sqs_filter(self, account, sqs, queues):
+        type = 'sqs queue'
+        unfiltered_queues = []
+        for q in queues:
+            if self.resource_has_special_name(account, type, q):
+                continue
+            # ignore queues with special tags
+            try:
+                tags = sqs.list_queue_tags(QueueUrl=q)['Tags']
+                if self.resource_has_special_tags(account, type, q, tags):
+                    continue
+                unfiltered_queues.append(q)
+            except KeyError as e:
+                if 'Tags' in e.message:
+                    unfiltered_queues.append(q)
+        return unfiltered_queues
+
+    def custom_dynamodb_filter(self, account, session, dynamodb, tables):
+        type = 'dynamodb table'
+        dynamodb_resource = session.resource('dynamodb')
+        unfiltered_tables = []
+        for t in tables:
+            if self.resource_has_special_name(account, type, t):
+                continue
+            # ignore queues with special tags
+            try:
+                table_arn = dynamodb_resource.Table(t).table_arn
+                tags = dynamodb.list_tags_of_resource(
+                    ResourceArn=table_arn)['Tags']
+                if self.resource_has_special_tags(account, type, t, tags):
+                    continue
+                unfiltered_tables.append(t)
+            except KeyError as e:
+                if 'Tags' in e.message:
+                    unfiltered_tables.append(t)
+        return unfiltered_tables
+
+    def custom_rds_filter(self, account, rds, instances):
+        type = 'rds instance'
+        unfiltered_instances = []
+        for i in instances:
+            if self.resource_has_special_name(account, type, i):
+                continue
+            # ignore queues with special tags
+            try:
+                instance = rds.describe_db_instances(DBInstanceIdentifier=i)
+                instance_arn = instance['DBInstances'][0]['DBInstanceArn']
+                tags = rds.list_tags_for_resource(
+                    ResourceName=instance_arn)['TagList']
+                if self.resource_has_special_tags(account, type, i, tags):
+                    continue
+                unfiltered_instances.append(i)
+            except KeyError as e:
+                print(i)
+                print(instance)
+                print(e.message)
+                if 'TagList' in e.message:
+                    unfiltered_instances.append(i)
+        return unfiltered_instances
+
+    def resource_has_special_name(self, account, type, resource):
+        skip_msg = '[{}] skipping {} '.format(account, type) + \
```

Comment:
I just did a test and better to use a word boundary `\b`:

https://gist.github.com/649b2cc36d8c12508a007beba25bfb73

Reply:
i used this suggestion: https://github.com/app-sre/qontract-reconcile/pull/84#discussion_r289324759

i would like the integration to not delete resources even if they have the word `prod` in the middle of the resource (or at any index for that matter) since this is a safety mechanism.

also pushed the `return True` under the `if`.

thanks!